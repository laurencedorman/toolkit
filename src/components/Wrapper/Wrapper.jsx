/**
 * Copyright (c) Colibri SAS - ManoMano
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from './Wrapper.module.scss';

/**
 * @visibleName Wrapper/Header/Footer
 */
const Wrapper = ({ children, className, direction, type, style, dataQa }) => {
  const classNames = cn(
    'toolkit',
    styles.wrapper,
    {
      [styles.row]: direction === 'row',
      [styles.column]: direction === 'column',
    },
    className
  );

  return React.createElement(
    type,
    {
      className: classNames,
      type,
      direction,
      style,
      'data-qa': dataQa,
    },
    children
  );
};

Wrapper.displayName = 'Wrapper';

Wrapper.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.func,
    PropTypes.element,
    PropTypes.elementType,
    PropTypes.node,
    PropTypes.arrayOf([
      PropTypes.element,
      PropTypes.elementType,
      PropTypes.node,
    ]),
  ]).isRequired,
  className: PropTypes.string,
  direction: PropTypes.string,
  style: PropTypes.shape({}),
  type: PropTypes.oneOf(['div', 'header', 'footer', 'section']),
  dataQa: PropTypes.string,
};

Wrapper.defaultProps = {
  className: '',
  direction: 'column',
  type: 'div',
  dataQa: '',
  style: {},
};

export default Wrapper;
