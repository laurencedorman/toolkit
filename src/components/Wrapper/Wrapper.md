Wrapper :

- choose <b>content direction</b> with props "<b>direction</b>" (column by default).
- choose <b>tag type</b> of wrapper "<b>div | header | footer</b>" (div by default).

```js
<Wrapper>
  <h4 className="h4">Simple Wrapper with display Flex and tag HTML type</h4>
</Wrapper>
```

```js
<Wrapper type="header">
  <h4 className="h4">
    Wrapper with <b>header</b> type
  </h4>
</Wrapper>
```

```js
<Wrapper type="footer">
  <h4 className="h4">
    Wrapper with <b>footer</b> type
  </h4>
</Wrapper>
```
