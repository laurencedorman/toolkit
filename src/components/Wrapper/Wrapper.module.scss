/**
 * Copyright (c) Colibri SAS - ManoMano
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

@import '../../styles/import.scss';

.wrapper {
  position: relative;
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  flex-flow: wrap;
  flex: 0 0 auto;
  width: 100%;
  padding: em($space);
  background-color: transparent;
  box-sizing: border-box;

  &.row {
    flex-direction: row;
  }
  &.column {
    flex-direction: column;
  }
}
