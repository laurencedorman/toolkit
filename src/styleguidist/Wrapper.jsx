/**
 * Copyright (c) Colibri SAS - ManoMano
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { PureComponent } from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import '../index.scss';

type propTypes = { children: Node }

export default class Wrapper extends PureComponent<propTypes> {
  render() {
    const { children } = this.props;

    return (
      <Router>{children}</Router>
    );
  }
}
