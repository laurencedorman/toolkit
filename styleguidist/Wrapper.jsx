/**
 * Copyright (c) Colibri SAS - ManoMano
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import '../src/index.scss';

const Wrapper = ({ children }) => {
  return <Router>{children}</Router>;
};

export default Wrapper;
