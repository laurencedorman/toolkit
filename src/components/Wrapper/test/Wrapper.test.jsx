/**
 * Copyright (c) Colibri SAS - ManoMano
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { shallow } from 'enzyme';
import Wrapper from '../Wrapper';

describe('Wrapper', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <Wrapper>
        <h1>children</h1>
      </Wrapper>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
