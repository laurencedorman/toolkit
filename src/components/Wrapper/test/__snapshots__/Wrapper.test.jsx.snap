// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper should match snapshot 1`] = `
<div
  className="toolkit wrapper column"
  data-qa=""
  direction="column"
  style={Object {}}
  type="div"
>
  <h1>
    children
  </h1>
</div>
`;
